src/gui/symbology/qgsstylemanagerdialog.cpp

```diff
@@ -518,18 +460,166 @@ QgsStyleManagerDialog::QgsStyleManagerDialog( QgsStyle *style, QWidget *parent,
   mSliderIconSize->setValue( thumbnailSize );
   connect( mSliderIconSize, &QSlider::valueChanged, this, &QgsStyleManagerDialog::setThumbnailSize );
   setThumbnailSize( thumbnailSize );
+}
+
+void QgsStyleManagerDialog::setCurrentStyle( QgsStyle *style )
+{
+  if ( mStyle == style )
+    return;
+
+  if ( mStyle )
+  {
+    disconnect( mStyle, &QgsStyle::symbolSaved, this, &QgsStyleManagerDialog::populateList );
+    disconnect( mStyle, &QgsStyle::groupsModified, this, &QgsStyleManagerDialog::populateGroups );
+    disconnect( mStyle, &QgsStyle::aboutToBeDestroyed, this, &QgsStyleManagerDialog::currentStyleAboutToBeDestroyed );
+  }
+
+  QgsCheckableStyleModel *oldModel = mModel;
+
+  mStyle = style;
+  const bool readOnly = isReadOnly();
+  if ( mStyle != QgsStyle::defaultStyle() )
+  {
+    if ( !mActionCopyToDefault )
+    {
+      mActionCopyToDefault = new QAction( tr( "Copy Selection to Default Style…" ), this );
```

Comment:
Would it be useful to have a Copy Selection to Project Style when the current style isn't the project style?

Reply:
I'll defer this to a follow up